src/raftstore/store/store.rs

```diff
@@ -280,36 +281,43 @@ impl<T, C> Store<T, C> {
                     try!(peer_storage::recover_from_applying_state(
                         &self.kv_engine,
                         &self.raft_engine,
+                        &raft_wb,
                         region_id
                     ));
-                }
-
-                let mut peer = try!(Peer::create(self, region));
-
-                if local_state.get_state() == PeerState::Applying {
                     applying_count += 1;
-                    info!(
-                        "region {:?} is applying in store {}",
-                        local_state.get_region(),
-                        self.store_id()
-                    );
-                    peer.mut_store().schedule_applying_snapshot();
+                    applying_regions.push(region.clone());
+                    return Ok(true);
                 }
 
+                let peer = try!(Peer::create(self, region));
                 self.region_ranges.insert(enc_end_key(region), region_id);
-                // No need to check duplicated here, because we use region id as the key
-                // in DB.
+                // No need to check duplicated here, because we use region id as the key in DB.
                 self.region_peers.insert(region_id, peer);
                 Ok(true)
             }
         ));
 
         if !kv_wb.is_empty() {
             self.kv_engine.write(kv_wb).unwrap();
+            self.kv_engine.flush_wal(true).unwrap();
         }
-
         if !raft_wb.is_empty() {
             self.raft_engine.write(raft_wb).unwrap();
+            self.raft_engine.sync_wal().unwrap();
+        }
+
+        // schedule applying snapshot after raft writebatch were written.
+        for region in applying_regions {
+            info!(
+                "region {:?} is applying in store {}",
```

Comment:
```
info!("{} region {:?} is applying", self.tag, region);
```

Reply:
already have "in store {}"

Reply:
no, we should follow the log format now.

Reply:
maybe another pr,  change all 'in store {}' to 'self.tag'